sqlglot/optimizer/qualify_columns.py

```diff
@@ -0,0 +1,321 @@
+import sqlglot.expressions as exp
+
+
+def qualify_columns(expression, schema):
+    """
+    Rewrite sqlglot AST to have fully qualified columns.
+
+    Example:
+        >>> import sqlglot
+        >>> schema = {"tbl": {"col": "INT"}}
+        >>> expression = sqlglot.parse_one("SELECT col FROM tbl")
+        >>> qualify_columns(expression, schema).sql()
+        'SELECT tbl.col AS col FROM tbl'
+
+    Args:
+        expression (sqlglot.Expression): expression to qualify
+        schema (dict): Mapping of table names to all available columns
+    Returns:
+        sqlglot.Expression: qualified expression
+    """
+    expression = expression.copy()
+    _qualify_union(expression, schema, Sequence(), {})
+    return expression
+
+
+class Sequence:
+    """
+    Auto-incrementing sequence.
+
+    Example:
+        >>> seq = Sequence()
+        >>> seq.next()
+        0
+        >>> seq.next()
+        1
+
+    This is useful for autogenerating unique alias names.
+    """
+
+    def __init__(self):
+        self.i = -1
+
+    def next(self):
+        self.i += 1
+        return self.i
+
+
+def _qualify_union(expression, schema, sequence, parent_selectables):
```

Comment:
the name is a bit misleading cuz it's union or select

Reply:
I'll change to _qualify_statement, to match the parser